tests/remote_transfer_services/dandi_transfer_tools.py

```diff
@@ -120,3 +122,21 @@ def test_staging_backward_compatibility(tmp_path):
         # Check that deprecation warning was issued
         deprecation_warnings = [warning for warning in w if issubclass(warning.category, DeprecationWarning)]
         assert len(deprecation_warnings) == 1, f"Expected 1 deprecation warning, got {len(deprecation_warnings)}"
+
+@pytest.mark.skipif(
+    not HAVE_EMBER_KEY,
+    reason="You must set your DANDI_API_KEY to run this test!",
+)
+def test_automatic_ember_upload(tmp_path):
+    nwb_folder_path = tmp_path / "test_nwb"
+    nwb_folder_path.mkdir()
+    metadata = get_default_nwbfile_metadata()
+    metadata["NWBFile"].update(
+        session_start_time=datetime.now().astimezone(),
+        session_id=f"test-automatic-upload-{sys.platform}-{get_python_version().replace('.', '-')}",
+    )
+    metadata.update(Subject=dict(subject_id="foo", species="Mus musculus", age="P1D", sex="U"))
+    with NWBHDF5IO(path=nwb_folder_path / "test_nwb_1.nwb", mode="w") as io:
+        io.write(make_nwbfile_from_metadata(metadata=metadata))
+
+    automatic_dandi_upload(dandiset_id="000431", nwb_folder_path=nwb_folder_path, instance="ember")
```

Comment:
@h-mayorquin FYI You will need to make an account on EMBER so I can add you to this Dandiset so you can generate an API key to set in the CI